packages/react/src/components/Breadcrumb/BreadcrumbItem.tsx

```diff
@@ -0,0 +1,140 @@
+import { usePrefix } from '../../internal/usePrefix';
+import { ForwardRefReturn } from '../../types/common';
+import { Text } from '../Text';
+
+export interface BreadcrumbItemProps
+  extends React.HTMLAttributes<HTMLLIElement> {
+  'aria-current'?: AriaAttributes['aria-current'];
+
+  /**
+   * Specify an optional className to be applied to the container node
+   */
+  className?: string;
+
+  /**
+   * Optional string representing the link location for the BreadcrumbItem
+   */
+  href?: string;
+
+  /**
+   * Provide if this breadcrumb item represents the current page
+   */
+  isCurrentPage?: boolean;
+}
+
+const BreadcrumbItem: ForwardRefReturn<HTMLLIElement, BreadcrumbItemProps> =
+  React.forwardRef(function BreadcrumbItem(
+    {
+      'aria-current': ariaCurrent,
+      children,
+      className: customClassName = '',
+      href,
+      isCurrentPage,
+      ...rest
+    }: PropsWithChildren<BreadcrumbItemProps>,
+    ref: React.Ref<HTMLLIElement>
+  ) {
+    const prefix = usePrefix();
+    const className = cx({
+      [`${prefix}--breadcrumb-item`]: true,
+      // We set the current class only if `isCurrentPage` is passed in and we do
+      // not have an `aria-current="page"` set for the breadcrumb item
+      [`${prefix}--breadcrumb-item--current`]:
+        isCurrentPage && ariaCurrent !== 'page',
+      [customClassName]: !!customClassName,
+    });
+
+    const child = children as React.FunctionComponentElement<any>;
+    if (
+      child.type &&
+      child.type.displayName !== undefined &&
+      child.type.displayName.includes('OverflowMenu')
+    ) {
+      const horizontalOverflowIcon = (
+        <OverflowMenuHorizontal className={`${prefix}--overflow-menu__icon`} />
+      );
+      return (
+        <li className={className} {...rest}>
+          {React.cloneElement(child, {
+            menuOptionsClass: `${prefix}--breadcrumb-menu-options`,
+            menuOffset: { top: 10, left: 59 },
+            renderIcon: () => horizontalOverflowIcon,
+          })}
+        </li>
+      );
+    }
+
+    if (typeof children === 'string') {
+      return (
+        <li className={className} ref={ref} {...rest}>
+          {href ? (
+            <Link href={href} aria-current={ariaCurrent}>
+              {children}
+            </Link>
+          ) : (
+            <Text className={`${prefix}--link`}>{children}</Text>
+          )}
+        </li>
+      );
+    }
+
+    return (
+      <li className={className} ref={ref} {...rest}>
+        {React.cloneElement(child, {
+          'aria-current': ariaCurrent,
+          className: cx(`${prefix}--link`, child.props.className),
+        })}
+      </li>
+    );
+  });
+
+BreadcrumbItem.displayName = 'BreadcrumbItem';
+
+BreadcrumbItem.propTypes = {
+  'aria-current': PropTypes.oneOfType([
+    PropTypes.bool,
+    PropTypes.oneOf([
+      'false',
+      'true',
+      'page',
+      'step',
+      'location',
+      'date',
+      'time',
+    ] as const),
+  ]),
```

Comment:
I'm not sure this warrants a breaking change either. [The spec](https://www.w3.org/TR/wai-aria-1.2/#aria-current) is quite clear on what values are allowed, though there is one exception (I bolded below for emphasis)

> The [aria-current](https://www.w3.org/TR/wai-aria-1.2/#aria-current) [attribute](https://www.w3.org/TR/wai-aria-1.2/#dfn-attribute) is a token type. **Any value not included in the list of allowed values SHOULD be treated by [assistive technologies](https://www.w3.org/TR/wai-aria-1.2/#dfn-assistive-technology) as if the value true had been provided.** If the attribute is not present or its value is an empty string or undefined, the default value of false applies and the [aria-current](https://www.w3.org/TR/wai-aria-1.2/#aria-current) [state](https://www.w3.org/TR/wai-aria-1.2/#dfn-state) MUST NOT be exposed by user agents or assistive technologies.

So technically any string is allowable and coalesces to `true`. You could argue these proptypes (plus the typings from React) are incorrect, but I don't think this presents much risk to have specific strings instead of allowing any string. If we get user feedback that some use case is missed, we can roll back this change very easily.